ui/packages/app/src/project_setting/metadata/Labels.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton
+} from "@elastic/eui";
+import { isDNS1123Label } from "../../validation/validation";
+import config from "../../config";
+
+export const Labels = ({ onChange, initialLabels }) => {
+  const [items, setItems] = useState([]);
+
+  useEffect(() => {
+    if (initialLabels) {
+      setItems(
+        initialLabels.map((label, idx) => ({
+          ...label,
+          idx,
+          isKeyValid: true,
+          isValueValid: true
+        }))
+      );
+    }
+  }, [initialLabels]);
```

Comment:
Do we need an effect for this? If you're just intending to set the values in `items` with those in the prop `initialLabels`, is it possible to just set them using `useState` in line 14? 

Also, it seems like most of the lines in this file are copied from `ui/packages/app/src/project_setting/form/Labels.js`. Is it possible to refactor that component (with the initial items stuff) and then import that component to use it directly?

Reply:
yes general comment, there are too many duplicates and copy-paste, please try to it keep more concise